src/Cli/dotnet/TransactionalAction.cs

```diff
@@ -65,19 +67,44 @@ public void Rollback(Enlistment enlistment)
             // This automatically inherits any ambient transaction
             // If a transaction is inherited, completing this scope will be a no-op
             T result = default(T);
-            using (var scope = new TransactionScope(
-                TransactionScopeOption.Required,
-                TimeSpan.Zero))
+            try
             {
-                Transaction.Current.EnlistVolatile(
-                    new EnlistmentNotification(commit, rollback),
-                    EnlistmentOptions.None);
+                using (var scope = CreateTransactionScope(
+                    TimeSpan.FromDays(10)))
+                {
+                    Transaction.Current.EnlistVolatile(
+                        new EnlistmentNotification(commit, rollback),
+                        EnlistmentOptions.None);
+
+                    result = action();
 
-                result = action();
+                    scope.Complete();
+                }
 
-                scope.Complete();
+                return result;
             }
-            return result;
+            catch (TransactionAbortedException ex)
+            {
+                Reporter.Verbose.WriteLine(string.Format("TransactionAbortedException Message: {0}", ex.Message));
+                Reporter.Verbose.WriteLine(
+                    $"Inner Exception Message: {ex?.InnerException?.Message + "---" + ex?.InnerException}");
+                throw;
+            }
+        }
+
+        private static void SetTransactionManagerField(string fieldName, object value)
+        {
+            typeof(TransactionManager).GetField(fieldName, BindingFlags.NonPublic | BindingFlags.Static)
+                .SetValue(null, value);
+        }
+
+        // HACK: https://github.com/dotnet/sdk/issues/21101
+        // we should replace TransactionScope later.
+        private static TransactionScope CreateTransactionScope(TimeSpan timeout)
+        {
+            SetTransactionManagerField("s_cachedMaxTimeout", true);
+            SetTransactionManagerField("s_maximumTimeout", timeout);
```

Comment:
These are static fields, so it probably makes sense to just set them once rather than each time we create a transaction.